src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -90,7 +91,8 @@ const ChoiceExerciseScreen = ({
       disciplineTitle,
       exercise: exerciseKey,
       results,
-      closeExerciseAction: route.params.closeExerciseAction
+      closeExerciseAction: route.params.closeExerciseAction,
+      unlockedNextExercise: progress[disciplineId] ? !progress[disciplineId]![exerciseKey] : true
```

Comment:
why is this dependent on the previous progress and not the current results? I find this quite hard to understand, can you explain? What are the possible values of `progress[disciplineId][exerciseKey]`?

Reply:
possible values are a score from 0 to 1 or undefined. With negation at the beginning we just check if it is undefined, which means the exercise was not done before, therefore is done the first time and therefore unlocks the next exercise.

Reply:
but negation in the beginning also returns true if the score is `0`, doesn't it? I think it should be explicitely `!== undefined`

Reply:
you are right. fixed it.